### ISSUE TITLE
`doc_cfg` does not have any effect on the re-exports

### ISSUE THREAD

**Issue by nagisa** [rust-lang/rust#85043]
Consider the following code:

```rust
#[doc(cfg(feature="include_file"))]
pub mod include_file {
    pub fn generated() {}
}

pub use include_file::*;
pub use include_file::generated;
pub use include_file::generated as renamed;
```

The produced documentation looks like this:

![An image of rustdoc output for the crate root, showing 3 reexports, `use include_file::*` which has a doc_cfg badge, and `use include_file::generated` as well as `use include_file::generated as renamed` which don't](https://user-images.githubusercontent.com/679122/117484130-323f7580-af56-11eb-8a0a-cf4b0a60eeb1.png)

I think we should be showing the doc_cfg badge consistently here: either on all of the re-exports or none of them. I would probably lean towards "none" in this particular case, in part because it would be easier to implement properly situations where re-exports don't share the features of the modules that are being re-exported. For instance:

```rust
#[doc(cfg(feature="include_file"))]
pub mod include_file {
    pub fn generated() {}
}
#[doc(cfg(feature="reexport_glob"))]
pub use include_file::*;
#[doc(cfg(feature="reexport_generated"))]
pub use include_file::generated;
#[doc(cfg(feature="reexport_renamed"))]
pub use include_file::generated as renamed;
```

which generates exactly the same documentation page as shown above. I believe in this case we should show exactly the specified `cfg` badges, rather than attempting to infer from what items are being re-exported.

**Comment by est31** [comment]
>  I believe in this case we should show exactly the specified cfg badges

I agree.

You can only use something from a module if the module is available, which means that if there are multiple implementations of a module gated by different cfg attributes, but the use is free of any cfg, there should be no inheritance from the module that rustdoc ran for. Conversely, if the pub use contains a more restricted cfg than the module, you would want to document that restricted cfg instead of the module's cfg, because the use would simply not be available in some cases.

TLDR: pub use should only show what attributes on the use itself contain.

**Comment by GuillaumeGomez** [comment]
I disagree with the reasoning here: if you don't enable the mentioned feature, the code will not compile, meaning you actually need to add the `cfg` on the reexports as well. So I don't think there is anything to be fixed in here.
